src/dmd/nogc.d

```diff
@@ -227,3 +244,24 @@ Expression checkGC(Scope* sc, Expression e)
     }
     return e;
 }
+
+/**
+ * Removes `HookTraceImpl` if found from `fd`.
+ * Parameters:
+ *  fd = The function declaration to remove `HookTraceImpl` from
+ */
+private FuncDeclaration stripHookTraceImpl(FuncDeclaration fd)
```

Comment:
Again, please explain why this functionality is needed for the next person that will have to read this code.